### ISSUE TITLE
Partially Succeeded Release appears as Successfull in Release Orchestrator

### ISSUE THREAD

**Issue by kboro** [dmitryserbin/azdev-release-orchestrator#40]
Hello,
When Release triggered by Rel Orchestrator is finished as "Partially Succeed" then Rel Orchestrator task shows it as "Succeeded"

Expecatation here is to mark Release Orchestrator task as "Partially Succeeded" when triggered Release finish as "Partially Succeeded"

**Comment by dmitryserbin** [comment]
Thanks for reporting this. I will have a look.

**Comment by Duke4848** [comment]
Having exactly the same issue and this is quite important as partially successful test run(pipeline) is shown a successful one which is very misleading. 
Would you please take a look at it?

**Comment by dmitryserbin** [comment]
Thanks @Duke4848, I feel your pain. Will try to address this soon.

**Comment by Duke4848** [comment]
> Thanks @Duke4848, I feel your pain. Will try to address this soon.
Awesome, thank you very much!
I've analysed your code and I think this part is the reason, however I have 0 experience in writing tasks for pipelines.
I've edited the code below, I think it will fail the stage if the orchestrated stage partially fails.
```
 // Get release status
    public getStatus(): ReleaseStatus {
        //[...]
        if (completed) {
            verbose(`All release stages completed`);
            // Get rejected or canceled stages
            const failed: boolean = this.progress.filter((i) =>
                i.status === ri.EnvironmentStatus.Rejected || i.status === ri.EnvironmentStatus.Canceled).length > 0
                || i.status === ri.EnvironmentStatus.PartiallySucceeded; //basing on line 1531 from https://github.com/microsoft/azure-devops-node-api/blob/master/api/interfaces/ReleaseInterfaces.ts
            if (failed) {
                result = ReleaseStatus.Failed;
            } else {
                result = ReleaseStatus.Succeeded;
            }
        //[...]
        return result;
    }
```

**Comment by dmitryserbin** [comment]
To be addressed in #45

**Comment by Duke4848** [comment]
Hi, thanks for addressing, however I've worked around it by forcing the orchestrated build to fail but not partially succeed. Just letting you know.
